pkg/session/session.go

```diff
@@ -40,18 +42,60 @@ import (
 	"github.com/vmware/govmomi/vim25/methods"
 	"github.com/vmware/govmomi/vim25/soap"
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
 
 	infrav1 "sigs.k8s.io/cluster-api-provider-vsphere/apis/v1beta1"
 	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/constants"
 )
 
+const (
+	metricNameSpace        = "session"
+	metricLabelServer      = "server"
+	metricLabelDC          = "dc"
+	metricLabelUsername    = "username"
+	metricLabelSessionKey  = "sessionKey"
+	metricLabelSessionKeys = "sessionKeys"
```

Comment:
I'm not sure about having a cache key or a list of keys as a label, it seems this is more debug info + it is somehow redundant with server/datacenter/user labels